core/src/main/java/org/infinispan/remoting/transport/jgroups/SingleSiteRequest.java

```diff
@@ -43,8 +51,11 @@ private void receiveResponse(Address sender, Response response) {
          T result;
          synchronized (responseCollector) {
             if (isDone()) {
-               throw new IllegalStateException(
-                     "Duplicate response received for x-site request " + requestId + " from " + sender);
+               if (log.isTraceEnabled()) {
```

Comment:
```suggestion
               if (log.isTraceEnabled() && response != CacheNotFoundResponse.INSTANCE) {
```

Actually I'm not sure this trace log is needed at all: even if we had a bug somewhere that caused a request to receive multiple responses, the first one would complete it, so this would only detect the very unlikely concurrent responses for the same request.

RequestRepository already logs a trace message when it doesn't find a request for a response, and that is much more likely to hit if we really are receiving multiple responses.

Removing the trace log would also remove the need for the `handleSiteUnreachableEvent` field.

Reply:
sure. With the fix in `org.infinispan.remoting.transport.jgroups.JGroupsTransport#siteUnreachable()` it should never happen anyway :+1: